sdk/queue/src/queue.ts

```diff
@@ -84,14 +84,19 @@ export async function syncQueue(
 
       status.running += 1
       return syncer(action, msg, ch)
-        .then((_v : any) => {
-          try {
-            ch.ack(msg)
-            status.running -= 1
-            return finalizeShutdown()
-          } catch (e) {
-            console.error("Could not ack a successful message! Action Id", action.actionId, e)  
-            throw e
+        .then(async (_v: any) => {
+          if (_v) {
+            try {
+              ch.ack(msg)
+              status.running -= 1
+              return finalizeShutdown();
+            } catch (e) {
+              console.error("Could not ack a successful message! Action Id", action.actionId, e)
+              throw e
+            }
+          } else {
+            await ch.nack(msg, false, false)
```

Comment:
ivana, you should call return finalizeShutdown(); after the nack too, it's a way to check if a shutdown has been requested elsewhere and to handle it properly

(probably not the best name for the function, but can't think of a good one right now)

Reply:
and check if ack and nack are promises or regular function, on one line you have await nack, on the other a simple call